### ISSUE TITLE
rootless podman not using rootless_storage_path from /etc/containers/storage.conf

### ISSUE THREAD

**Issue by rcasta74** [containers/podman#7546]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Not sure if bug or not well documented behaviour. From `man containers-storage.conf`:
>        rootless_storage_path="$HOME/.local/share/containers/storage"
>          Storage path for rootless users. By default the graphroot for rootless users is set to  $XDG_DATA_HOME/containers/storage,  if  XDG_DATA_HOME  is
>        set.   Otherwise  $HOME/.local/share/containers/storage  is used.  This field can be used if administrators need to change the storage location for
>        all users.
The last sentence let me think I can define it somewhere to override the default path for all users (usually system wide configurations are defined into /etc).
I tried to define it into `/etc/containers/storage.conf` as:
`rootless_storage_path = "/mnt/containers/storage-$UID"`
but running the command `podman info` as rootless user (after removing folders `~/.config/containers` and `~/.local/share/containers`) I still get the default path `$HOME/.local/share/containers/storage`
To have the folder changed I need to properly create the file `~/.config/containers/storage.conf` for each users.

**Steps to reproduce the issue:**

1. Define a new path for rootless storage in `/etc/containers/storage.conf`
`rootless_storage_path = "/mnt/containers/storage-$UID"`

2. Clean containers folder as normal user (not root)
`rm -fr ~/.config/containers ~/.local/share/containers`

3. Check new storage path with the same user of above command
`podman info | fgrep graphRoot`

**Describe the results you received:**
`  graphRoot: /home/rcastagnola/.local/share/containers/storage`

**Describe the results you expected:**
`  graphRoot: /mnt/containers/storage-1001`

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      2.0.6
API Version:  1
Go Version:   go1.14.6
Built:        Tue Sep  1 21:26:51 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.1
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.19-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.19, commit: 5dce9767526ed27f177a8fa3f281889ad509fea7'
  cpus: 4
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.4-200.fc32.x86_64
  linkmode: dynamic
  memFree: 1366425600
  memTotal: 8182431744
  ociRuntime:
    name: crun
    package: crun-0.14.1-3.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.14.1
      commit: 598ea5e192ca12d4f6378217d3ab1415efeddefa
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1001/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-1.fc32.x86_64
    version: |-
      slirp4netns version 1.1.4
      commit: b66ffa8e262507e37fca689822d23430f3357fe8
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 8212967424
  swapTotal: 8271163392
  uptime: 4h 47m 21.65s (Approximately 0.17 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/rcastagnola/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.2-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/rcastagnola/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/1001/containers
  volumePath: /home/rcastagnola/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 1598988411
  BuiltTime: Tue Sep  1 21:26:51 2020
  GitCommit: ""
  GoVersion: go1.14.6
  OsArch: linux/amd64
  Version: 2.0.6
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.0.6-1.fc32.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by Luap99** [comment]
Yes this is a bug. See: #7513

**Comment by vrothberg** [comment]
@rhatdan PTAL

**Comment by rhatdan** [comment]
This should have been fixed in storage here https://github.com/containers/storage/pull/713/commits/3c3285331d220e5571a82f96444db8e3e0c6a875
Which has been vendored into master.

**Comment by rhatdan** [comment]
Should be fixed in 2.1
